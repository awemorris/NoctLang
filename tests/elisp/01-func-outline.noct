func foo(a, b, c) {
    if (a == 0) {
        i = 0;
    } else if (a == 1) {
        i = 1;
    } else {
        i = 2;
    }

    var arr = [1, 2, 3];
    for (v in arr) {
        print(v);
    }

    for (j in 0..10) {
        i = j;
    }

    for (k, v in {aaa: "bbb"}) {
        i = k;
    }

    while (i == 0) {
        j = 0;
    }
}
