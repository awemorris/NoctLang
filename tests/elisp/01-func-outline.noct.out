(defun foo (a b c)
  (let ((k 0) (j 0) (v 0) (arr 0))
    (catch 'early-return
      (if (= a 0)
        (progn
          (setq i 0)
        )
        (if (= a 1)
          (progn
            (setq i 1)
          )
          (progn
            (setq i 2)
          )
        )
      )
      (setq arr '(1 2 3))
      (dolist (v arr)
        (progn
          (print v)
        )
      )
      (let ((j 0))
        (while (< j 10)
          (progn
            (setq i j)
            (setq j (+ 1 j))
          )
        )
      )
      (dolist (pair '((aaa . "bbb")))
        (let ((k (car pair))
              (v (cdr pair)))
          (setq i k)
        )
      )
      (while (= i 0)
        (progn
          (setq j 0)
        )
      )
      0
    )
  )
)
